### PR TITLE
User new events to add the facerecognition tab

### PR DESCRIPTION
**Pull request by matiasdelellis**
Issue #257 

At this point I could only load our scripts with the sidebar.. still returns errors:

> TypeError: OCA.Files.DetailTabView is undefined

It is the same error of activity, comments, and version tabs.

I'm not sure if I should migrate to VUE, or is there an alternative.

### CODE REVIEW

File: lib/Listener/LoadSidebarListener.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	public function handle(Event $event): void {
+		if (!($event instanceof LoadSidebar)) {
+			return;
+		}
+
+		\OCP\Util::addScript('facerecognition', 'vendor/lozad');
+		\OCP\Util::addScript('facerecognition', 'fr-dialogs');
+		\OCP\Util::addScript('facerecognition', 'files-tabview');
+		\OCP\Util::addScript('facerecognition', 'files-plugin');
+		\OCP\Util::addStyle('facerecognition', 'files-tabview');
```
Comment (skjnldsv) [inline lib/Listener/LoadSidebarListener.php:45]:
```suggestion
		Util::addScript('facerecognition', 'vendor/lozad');
		Util::addScript('facerecognition', 'fr-dialogs');
		Util::addScript('facerecognition', 'files-tabview');
		Util::addScript('facerecognition', 'files-plugin');
		Util::addStyle('facerecognition', 'files-tabview');
```

Comment (matiasdelellis) [reply lib/Listener/LoadSidebarListener.php:45]: Ok.. @skjnldsv
Thanks for all the answers .. I will try to migrate to vue.. :wink:

Comment (skjnldsv) [reply lib/Listener/LoadSidebarListener.php:45]: I might add the proper global standard for 20 though :see_no_evil:

Comment (matiasdelellis) [reply lib/Listener/LoadSidebarListener.php:45]: imagen

I dont have any idea what I'm doing .. but at least it shows something .. hahah

**Comment by matiasdelellis** [comment]
imagen

Wow.. Is working!. :smile:

File: lib/AppInfo/Application.php [hunk 2]
```diff
@@ -47,6 +51,7 @@ public function __construct(array $urlParams = []) {
 
 		$this->connectWatcher();
 		$this->connectSearch();
+		$this->addServiceListenders();
```
Comment (stalker314314) [inline lib/AppInfo/Application.php:54]: Listenders->Listeners maybe

**Comment by matiasdelellis** [comment]
@stalker314314 
I'm going to merge this change.. Beyond the increase in size, (From nothing to 145kb :sweat_smile:), the problem is that this depends on NC18.
My idea was to abandon NC16, but not NC17. Do you have a problem with this? :disappointed:

**Comment by matiasdelellis** [comment]
> Congratulation Matias, you really rock! (I really don't have any other meaningful comment to add:D)

haha. Thanks, but all the background task is thanks to you!. :smile:

**Pull request merged**